### PR TITLE
feat(server): Explicit retry of individual configs [INGEST-1220]

### PR DESCRIPTION
**Pull request by untitaker**
Add ability for Sentry to ask Relay to retry individual projectconfigs indefinetly. We will populate this new response attribute in case of exceptions happening in `get_project_config`. The corresponding PR in Sentry for this will be https://github.com/getsentry/sentry/pull/25451

The original plan was to simply omit the key from the response, however omitting the key is already very overloaded for other purposes such as relays not being trusted.

Note that since our backoff is global, it is still reset and the retry happens every `batch_interval`, not exponentially. We will just have to see if this is good enough, we will add another killswitch to Sentry that stops requesting retries from Relay.

### CODE REVIEW

File: tests/integration/fixtures/mini_sentry.py [hunk 1]
```diff
@@ -331,6 +331,7 @@ def fail(e):
 
     def reraise_test_failures():
         if sentry.test_failures:
+            return
             pytest.fail(
                 "{n} exceptions happened in mini_sentry:\n\n{failures}".format(
                     n=len(sentry.test_failures), failures=sentry.format_failures()
```
Comment (RaduW) [inline tests/integration/fixtures/mini_sentry.py]: something seems fishy here.

File: tests/integration/fixtures/mini_sentry.py [hunk 2]
```diff
@@ -331,6 +331,7 @@ def fail(e):
 
     def reraise_test_failures():
         if sentry.test_failures:
+            return
```
Comment (untitaker) [inline tests/integration/fixtures/mini_sentry.py]:
```suggestion
```

File: tests/integration/fixtures/mini_sentry.py (hunk 1 above)
Comment (untitaker) [reply tests/integration/fixtures/mini_sentry.py]: fixed

**Review by RaduW — APPROVED** [review]
Looks good.

The thing that worries me a bit is that a project_key that is configured for retry will retry forever if sentry is not able to provide the config( is that what we want?).

**Comment by untitaker** [comment]
@RaduW for that sentry needs to respond with "please retry" forever though. If `retryConfigs` is blank/empty the cycle ends.

**Comment by untitaker** [comment]
@jan-auer's feedback:

1) in a future version, sentry and relay may emit `retryConfig: [a]` if projectconfig for `a` takes too long to compute/can't be served from cache
2) for that to work, relay requests should indicate that they support retries, because we can't run with this behavior for requests from old relays
3) retryConfigs can be a bool flag on projectstate, similar to invalid/disabled

**Comment by jan-auer** [comment]
Superseded by https://github.com/getsentry/relay/pull/1263

**Pull request closed without merging**